extensions/wikia/Discussions/controllers/SpecialForumRedirectController.class.php

```diff
@@ -4,11 +4,162 @@ class SpecialForumRedirectController extends WikiaSpecialPageController {
+		$this->response->redirect( $discussionUrl );
+	}
+
+	/**
+	 * Redirect requests for the board page to discussions, e.g, from:
+	 *
+	 *   http://garth.wikia.com/wiki/Board:Some_Board_Name
+	 *
+	 * to:
+	 *
+	 *   http://garth.wikia.com/d/f?catId=2016136434548737693
+	 *
+	 * This is handled by registering a namespace controller (deprecated currently, but how wall
+	 * works) e.g. this line in Discussions.setup.php:
+	 *
+	 * $app->registerNamespaceController(
+	 *     NS_WIKIA_FORUM_BOARD,
+	 *     'SpecialForumRedirectController',
+	 *     'redirectBoardToCategory',
+	 * );
+	 *
+	 */
+	public function redirectBoardToCategory() {
+		// No template for this, we're only interested in setting the redirect.
+		$this->skipRendering();
+
+		$boardId = $this->getBoardId();
+		$redirectUrl = $this->legacyRedirect->getBoardRedirect( $boardId );
+
+		if ( empty( $redirectUrl ) ) {
+			// If there's any problem, just redirect to main discussion page
+			$redirectUrl = self::DISCUSSIONS_LINK;
+		}
+
+		$this->response->redirect( $redirectUrl );
+	}
+
+	/**
+	 * Redirect requests for a thread to discussions, e.g. from:
+	 *
+	 *   http://garth.garth.wikia-dev.us/wiki/Thread:2892
+	 *
+	 * to:
+	 *
+	 *   http://garth.wikia.com/d/p/2914856863801542335
+	 *
+	 * @param Title $thread
+	 */
+	public static function redirectThreadToPost( Title $thread ) {
+		$threadId = $thread->getArticleID();
+
+		$legacyRedirect = new LegacyRedirect( F::App()->wg->CityId );
+		$redirectUrl = $legacyRedirect->getThreadRedirect( $threadId );
+
+		if ( empty( $redirectUrl ) ) {
+			// If there's any problem, just redirect to main discussion page
+			$redirectUrl = self::DISCUSSIONS_LINK;
+		}
+
+		F::app()->wg->Out->redirect( $redirectUrl );
+	}
+
+	private function getDiscussionUrl() {
+		return self::DISCUSSIONS_LINK;
+	}
+
+	private function getBoardId() {
+		return $this->wg->Title->getArticleID();
+	}
+
+	/**
+	 * @param Article $article
+	 *
+	 * @return bool
+	 */
+	public static function onBeforePageHistory( &$article ) {
+		self::redirectPage( $article );
+		return true;
+	}
+
+	/**
+	 * @param Article $article
+	 * @param bool $outputDone
+	 * @param bool $useParserCache
+	 *
+	 * @return bool
+	 */
+	public static function onArticleViewHeader( &$article, &$outputDone, &$useParserCache ) {
+		self::redirectPage( $article );
+		return true;
+	}
+
+	private static function redirectPage( &$article ) {
+		$mainTitle = self::getMainTitle( $article );
+		if ( empty( $mainTitle ) ) {
+			return;
+		}
+
+		// Ignore non-forum board threads.  This allows MessageWall to display properly
+		if ( !$mainTitle->inNamespace( NS_WIKIA_FORUM_BOARD_THREAD ) ) {
+			return;
+		}
+
+		// Redirect to discussions
+		self::redirectThreadToPost( $mainTitle );
+	}
+
+	/**
+	 * Some of this logic has been borrowed from WallHooksHelper::onArticleViewHeader
+	 *
+	 * @param Article $article
+	 * @return null|Title
+	 */
+	private static function getMainTitle( Article $article ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)